### PR TITLE
[Focused Improvements] .eslintrc updates

### PR DESCRIPTION
**Pull request by CobyPear**
Restructured the .eslintrc files so that the packages are extending the configs from the root of the monorepo.
React and TypeScript specific configs have also been pulled out into their own config files which now live in `./eslint-configs`. This should make future maintenance a tad easier.

## Description / Motivation

general improvements to the monorepo

## Testing Details

linted each package and addressed the errors

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update (non-breaking change; modified files are limited to the `/docs` directory)
- [x] Technical debt/maintenance

### CODE REVIEW

File: packages/sitecore-jss-react-native/.eslintrc [hunk 1]
```diff
@@ -1,19 +1,12 @@
...
-    "ecmaFeatures": {
-      "jsx": true
-    }
-  },
+  "extends": [
+    "../../.eslintrc",
+    "../../eslint-configs/typescript"
+  ],
+  "plugins": [
+    "react-native",
```
Comment (ambrauer) [inline packages/sitecore-jss-react-native/.eslintrc:7]: Remove trailing comma
```suggestion
    "react-native"
```

**Review by ambrauer — CHANGES_REQUESTED** [review]
This is looking good, see some specific comments.

File: packages/sitecore-jss-react-forms/src/components/field-templates/field-validation-errors.tsx [hunk 2]
```diff
@@ -7,7 +7,7 @@ const FieldValidationErrors: React.FunctionComponent<LabelProps> = (props) => {
 
     // strip the errors component from the custom component props
     // (prevents infinite loop rendering if someone reuses this component as a custom error component)
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    // eslint-disable-next-line no-unused-vars
```
[the remark this answers is not in the document]
Comment (CobyPear) [reply packages/sitecore-jss-react-forms/src/components/field-templates/field-validation-errors.tsx:10]: Hmm I think I need to fix the way in which I'm testing changes locally, because it was giving me errors with the `@typescript-eslint/` and none without. I'll figure that out and revert these.

File: .eslintrc [hunk 3]
```diff
@@ -63,12 +53,33 @@
...
+    ]
+  },
+  "ignorePatterns": [
+    "src/test/*",
+    "src/tests/*",
+    "src/testData/*",
+    "src/*.test.*",
+    "src/**/*.spec.*",
+    "src/**/*.test.*",
+    "src/**/*.directive.*"
```
[the remark this answers is not in the document]
Comment (CobyPear) [reply .eslintrc:83]: The reason for this was because I saw `sitecore-jss-react-native` doing this and it made sense to keep everything the same, and `it` and `describe` were throwing linting errors. No problem reverting this!

File: .eslintrc [hunk 4]
```diff
@@ -29,31 +32,18 @@
...
     ],
-    "@typescript-eslint/typedef": "error",
-    "@typescript-eslint/type-annotation-spacing": "error",
-    "@typescript-eslint/semi": "error",
-    "@typescript-eslint/no-var-requires": "off",
-    "no-useless-escape": "off",
-    "spaced-comment": "error",
-    "curly": ["error", "multi-line"],
-    "eol-last": ["error", "always"],
+    "linebreak-style": "off",
```
[the remark this answers is not in the document]
Comment (CobyPear) [reply .eslintrc:46]: Yes! I think I wasn't getting the syntax for this right so I turned it off 😅 fixed.

File: eslint-configs/typescript.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        },
+      },
+    ],
+    '@typescript-eslint/member-ordering': 'error',
+    '@typescript-eslint/no-use-before-define': ['error', { functions: false, variables: false }],
+    '@typescript-eslint/typedef': 'error',
+    '@typescript-eslint/type-annotation-spacing': 'error',
+    '@typescript-eslint/semi': 'error',
+    '@typescript-eslint/no-var-requires': 'off',
+    '@typescript-eslint/no-explicit-any': 'off',
```
[the remark this answers is not in the document]
Comment (CobyPear) [reply eslint-configs/typescript.js:21]: Yup, lost in the shuffle, good catch.

File: eslint-configs/typescript.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
+module.exports = {
+  extends: ['plugin:@typescript-eslint/recommended', '../.eslintrc'],
```
[the remark this answers is not in the document]
Comment (CobyPear) [reply eslint-configs/typescript.js]: Makes sense. Done.

File: .eslintrc (hunk 3 above)
[the remark this answers is not in the document]
Comment (ambrauer) [reply .eslintrc:83]: Yeah, I guess my vote would be to keep them linted if they were previously (can keep the exception in react native).

**Review by ambrauer — APPROVED** [review]

File: .eslintrc (hunk 3 above)
[the remark this answers is not in the document]
Comment (CobyPear) [reply .eslintrc]: Alright, done!

File: .eslintrc (hunk 4 above)
[the remark this answers is not in the document]
Comment (mksby) [reply .eslintrc]: > Since we're forcing windows-style line endings in `.gitattributes`, should we enable that here?

Why is windows-style forced? JSS is supposed to be a cross-platform solution
